paho/extensions/rpc/rpc.go

```diff
@@ -68,13 +71,19 @@ func (h *Handler) Request(ctx context.Context, pb *paho.Publish) (*paho.Publish,
 	pb.Properties.ResponseTopic = fmt.Sprintf("%s/responses", h.c.ClientID)
 	pb.Retain = false
 
-	_, err := h.c.Publish(ctx, pb)
+	_, err = h.c.Publish(ctx, pb)
 	if err != nil {
 		return nil, err
 	}
 
-	resp := <-rChan
-	return resp, nil
+	select {
+	case <-time.After(defaultRequestTimeout):
```

Comment:
I would argue that providing a context here should be sufficient (if the user wants a timeout then they can set one, otherwise there is no timeout). Setting 10 seconds seems arbitrary (say the RPC job takes a minute to complete; the user has no way to turn this timeout off).

Reply:
thanks @MattBrittan, I have set the requestTimeout as an optional config